src/debugger/sdfg_python_runtime.ts

```diff
@@ -201,12 +130,10 @@ export class SdfgPythonDebuggerRuntime extends EventEmitter {
                 }
             });
         } else {
-            this.checkSdfgDirty(editor.document, uri).then(() => {
-                this.checkHasWrapper(editor, uri).then(() => {
-                    this.run(editor.wrapperFile, uri);
-                }, () => {
-                    this.sendEvent('end');
-                });
+            if (editor.document.isDirty)
+                editor.document.save();
```

Comment:
I would say not, because this is how VSCode handles other launches.
If autosave is on this would never be necessary, since the file shoulnd't remain dirty for more than a few milliseconds at a time. So what VSCode does is that if you execute / launch a program, and you do NOT have autosave turned on, it saves the document so you're actually executing what you wrote. (e.g. Python files)